### PR TITLE
US002 Creating Items

### PR DESCRIPTION
**Pull request by kuzivka**
Added theme for styling MUI components, created input  and modal components, implemented logic for task adding

### CODE REVIEW

File: src/components/AddNewTodoModal.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { useState, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { TextField, Modal, Box, Button, Typography } from '@mui/material';
+import { addTodo } from '../actions/actionCreators';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { LocalizationProvider, DateTimePicker } from '@mui/x-date-pickers';
+import { getNewTaskObject } from '../utils/getNewTaskObject';
+
+const style = {
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx]: Create separate file for styles

File: src/components/AddNewTodoModal.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  handleClose: () => void;
+}
+
+export function AddNewTodoModal(props: IAddNewTodoModalProps) {
+  const { open, handleClose } = props;
+  const [taskValue, setTaskValue] = useState('');
+  const [expirationDate, setExpirationDate] = useState(currentDate);
+  const dispatch = useDispatch();
+
+  const taskValueChangeHandler = useCallback(
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx:23]: You can import {ChangeEvent} from 'react', and use without React.

File: src/components/AddNewTodoModal.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          <DateTimePicker
+            minDateTime={currentDate}
+            renderInput={(props) => <TextField {...props} />}
+            label="Expiration  Date"
+            ampm={false}
+            inputFormat="dd/MM/yyyy HH:mm"
+            value={expirationDate}
+            onChange={(newValue: Date | null) => {
+              if (newValue) {
+                console.log(newValue);
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx]: remove console.log

Comment (Yuliia143) [reply src/components/AddNewTodoModal.tsx]: create separate function and use it, try to keep jsx clean

File: src/components/AddNewTodoModal.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                setExpirationDate(newValue);
+              }
+            }}
+          />
+        </LocalizationProvider>
+        <Box sx={{ display: 'flex', justifyContent: 'flex-end' }}>
+          <Button disabled={taskValue.trim() === ''} onClick={createNewTask}>
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx]: !taskValue.trim() , because empty string is false

File: src/components/AddNewTodoModal.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            value={expirationDate}
+            onChange={(newValue: Date | null) => {
+              if (newValue) {
+                console.log(newValue);
+                setExpirationDate(newValue);
+              }
+            }}
+          />
+        </LocalizationProvider>
+        <Box sx={{ display: 'flex', justifyContent: 'flex-end' }}>
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx]: create class in file with styles, try to not use inline styles

File: src/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+export default function Header() {
+  return (
+    <div>
+      <Typography sx={{ color: '#2b2b2b' }} variant="h1">
+        To Do
+      </Typography>
+      <Typography
+        variant="button"
+        sx={{
```
Comment (Yuliia143) [inline src/components/Header.tsx]: the same, separate file for styles

File: src/components/TaskCards.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import { Checkbox, List, ListItem, ListItemText } from '@mui/material';
+import { useSelector } from 'react-redux';
+import { getTaskListToShowSelector } from '../selectors/getTaskListToShowSelector';
+
+export default function TaskCards() {
+  const tasks = useSelector(getTaskListToShowSelector);
+
+  return (
+    <List
+      sx={{ width: '600px', maxWidth: '80vw', bgcolor: 'grey', margin: 'auto' }}
```
Comment (Yuliia143) [inline src/components/TaskCards.tsx]: the same x2

File: src/components/TaskCards.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    >
+      {tasks.map((task) => (
+        <ListItem
+          key={task.id}
+          sx={{
```
Comment (Yuliia143) [inline src/components/TaskCards.tsx]: the same x3

File: src/components/TaskInput.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+import { addTodo } from '../actions/actionCreators';
+import { getNewTaskObject } from '../utils/getNewTaskObject';
+import { AddNewTodoModal } from './AddNewTodoModal';
+
+export default function TaskInput() {
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [taskValue, setTaskValue] = useState('');
+  const dispatch = useDispatch();
+
+  const handleModalOpen = useCallback(() => {
```
Comment (Yuliia143) [inline src/components/TaskInput.tsx]: You can create one function and pass true/false, or make setModalOpen(!isModalOpen) in function

File: src/components/TaskInput.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    setModalOpen(false);
+  }, []);
+
+  const taskValueChangeHandler: ChangeEventHandler<HTMLInputElement> =
+    useCallback((event) => {
+      setTaskValue(event.target.value);
+    }, []);
+
+  const enterClickHandler = useCallback(
+    (event: any) => {
```
Comment (Yuliia143) [inline src/components/TaskInput.tsx]: Why type is any ?

File: src/components/TaskInput.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        dispatch(addTodo(getNewTaskObject(taskValue)));
+        setTaskValue('');
+      }
+    },
+    [dispatch, taskValue]
+  );
+
+  return (
+    <div
+      style={{
```
Comment (Yuliia143) [inline src/components/TaskInput.tsx]: the same x4

File: src/components/AddNewTodoModal.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+const currentDate = new Date();
```
Comment (Kost927) [inline src/components/AddNewTodoModal.tsx]: Do you need this variable in the global scope?

File: src/components/TaskCards.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+export default function TaskCards() {
+  const tasks = useSelector(getTaskListToShowSelector);
+
+  return (
+    <List
+      sx={{ width: '600px', maxWidth: '80vw', bgcolor: 'grey', margin: 'auto' }}
+    >
+      {tasks.map((task) => (
+        <ListItem
+          key={task.id}
```
Comment (Kost927) [inline src/components/TaskCards.tsx]: you can use destructuring for task: tasks.map(( { id, task, expiresAt } ) => ...

File: src/components/TaskCards.tsx [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+          sx={{
+            border: '1px solid #ddd',
+            borderBottom: 'none',
+            '&:last-child': { borderBottom: '1px solid #ddd' },
+          }}
+        >
+          <Checkbox edge="start" onChange={() => void 1} checked={false} />
+          <ListItemText
+            primary={task.task}
+            secondary={`Expires at: ${task.expiresAt}`}
```
Comment (Kost927) [inline src/components/TaskCards.tsx]: destructuring

File: src/components/AddNewTodoModal.tsx (hunk 12 above)
Comment (kuzivka) [reply src/components/AddNewTodoModal.tsx]: no, I don't)

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**